storage/innobase/row/row0ins.cc

```diff
@@ -2703,20 +2703,42 @@ row_ins_clust_index_entry_low(
 
 	DBUG_EXECUTE_IF("row_ins_row_level", goto skip_bulk_insert;);
 
-	if (!(flags & BTR_NO_UNDO_LOG_FLAG)
-	    && page_is_empty(block->page.frame)
-	    && !entry->is_metadata() && !trx->duplicates
-	    && !trx->check_unique_secondary && !trx->check_foreigns
-	    && !trx->dict_operation
+	if (((!(flags & BTR_NO_UNDO_LOG_FLAG)
+	      && page_is_empty(block->page.frame)
+	      && !entry->is_metadata() && !trx->duplicates
+	      && !trx->check_unique_secondary && !trx->check_foreigns
+	      && !trx->dict_operation
+	      && !index->table->skip_alter_undo
+	      && !index->table->is_active_ddl())
+	      || (index->table->skip_alter_undo
```

Comment:
The indentation looks misleading here. The `||` should be indented by 1 less space because it is at a higher level than the rest.

But, also the starting parenthesis that is being closed before the `||` looks possibly misplaced. Do we really need to evaluate `index->table->skip_alter_undo` if the `flags` bit is set, or the page is not empty, or it is not the index root page?

Please try to find the minimum necessary change to this `if` expression.